### ISSUE TITLE
Support schema files with an array of schemas inside

### ISSUE THREAD

**Issue by mdlavin** [lifeomic/json-schema-to-graphql-types#20]
The converter should handle arrays of JSON schems

**Comment by HavenLin** [comment]
I have got error as "{ UnknownTypeReference: The referenced type #/definitions/metrics (undefined) is unknown in http://******/*****_Response.json.metrics".

does this tool support "$ref" syntax?

the schema example as :

{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "id": "http://*********/*******_Response.json",
  "description": "A json-schema for ***** Response.",
  "definitions": {
    "metrics": {
      "type": "array",
      "items": {
        "type": "object",
        "properties": {
          "type": {
            "type": "string",
            "enum": [
              "A",
              "B"
            ]
          },
          "name": {
            "type": "string"
          }
        },
        "required": [
          "type",
          "name"
        ]
      },
      "default": null
    }
  },
  "type": "object",
  "properties": {
    "success": {
      "type": "boolean"
    },
    "metrics": {
      "$ref": "#/definitions/metrics"
    }
  }
}

**Comment by mdlavin** [comment]
@HavenLin yes, I expect that $refs should work since #16 was merged. One thing that stands out to me about your schema is that it has an 'id' that will not be converted nicely into a GraphQL type name. Since your issue is unrelated to array support in the issue, can you create a new issue with your details moved over?

@MorganHuegel here is another great opportunity to contribute. Can you:
1.  Create a testcase / PR to improve the validation of schema ids to make sure they are valid GraphQL type names?
2. Help Haven understand why his schema won't convert?

**Comment by HavenLin** [comment]
@mdlavin thank you for quickly response. that's very cool if it possible supports this feature. I will create new issue for this
